### PR TITLE
Use concurrency when creating all indicators

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description
Utilize `asyncio` and semaphores to execute the `create_indicator` function
concurrently when all indicators are created (`create_all_indicators`).

Adresses in part  #152

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by matthiasschaub** [comment]
Can not reproduce CI error locally:

```
VCR_RECORD_MODE=none pytest test_oqt.py
```

Error:

```bash
ERROR    asyncio:base_events.py:1707 unhandled exception during asyncio.run() shutdown

task: <Task finished name='Task-437' coro=<create_all_indicators.<locals>._create_indicator() done, defined at /opt/oqt/ohsome_quality_analyst/oqt.py:236> exception=CannotOverwriteExistingCassetteException("Can't overwrite existing cassette ('/opt/oqt/tests/integrationtests/fixtures/vcr_cassettes/mock.json') in your current record mode ('none').\nNo match for the request (<Request (POST) https://api.ohsome.org/v1/elements/count>) was found.\nNo similar requests, that have not been played, found.")>

Traceback (most recent call last):

  File "/usr/local/lib/python3.8/asyncio/runners.py", line 44, in run

    return loop.run_until_complete(main)

  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete

    return future.result()

  File "/opt/oqt/ohsome_quality_analyst/oqt.py", line 263, in create_all_indicators

    return await asyncio.gather(*tasks)

  File "/opt/oqt/ohsome_quality_analyst/oqt.py", line 238, in _create_indicator

    await create_indicator(

  File "/opt/oqt/ohsome_quality_analyst/oqt.py", line 176, in create_indicator

    await from_scratch()

  File "/opt/oqt/ohsome_quality_analyst/oqt.py", line 144, in from_scratch

    await indicator.preprocess()

  File "/opt/oqt/ohsome_quality_analyst/indicators/ghs_pop_comparison_roads/indicator.py", line 54, in preprocess

    query_results = await ohsome_client.query(

  File "/opt/oqt/ohsome_quality_analyst/ohsome/client.py", line 34, in query

    return await query_ohsome_api(url, data)

  File "/opt/oqt/ohsome_quality_analyst/ohsome/client.py", line 43, in query_ohsome_api

    resp = await client.post(url, data=data, headers=headers)

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/httpx/_client.py", line 1624, in post

    return await self.request(

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/httpx/_client.py", line 1361, in request

    response = await self.send(

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/vcr/stubs/httpx_stubs.py", line 124, in _async_vcr_send

    vcr_request, response = _shared_vcr_send(cassette, real_send, *args, **kwargs)

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/vcr/stubs/httpx_stubs.py", line 79, in _shared_vcr_send

    raise CannotOverwriteExistingCassetteException(cassette=cassette, failed_request=vcr_request)

vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/opt/oqt/tests/integrationtests/fixtures/vcr_cassettes/mock.json') in your current record mode ('none').

No match for the request (<Request (POST) https://api.ohsome.org/v1/elements/length>) was found.

No similar requests, that have not been played, found.


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/opt/oqt/ohsome_quality_analyst/oqt.py", line 238, in _create_indicator

    await create_indicator(

  File "/opt/oqt/ohsome_quality_analyst/oqt.py", line 176, in create_indicator

    await from_scratch()

  File "/opt/oqt/ohsome_quality_analyst/oqt.py", line 144, in from_scratch

    await indicator.preprocess()

  File "/opt/oqt/ohsome_quality_analyst/indicators/mapping_saturation/indicator.py", line 44, in preprocess

    query_results = await ohsome_client.query(

  File "/opt/oqt/ohsome_quality_analyst/ohsome/client.py", line 34, in query

    return await query_ohsome_api(url, data)

  File "/opt/oqt/ohsome_quality_analyst/ohsome/client.py", line 43, in query_ohsome_api

    resp = await client.post(url, data=data, headers=headers)

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/httpx/_client.py", line 1624, in post

    return await self.request(

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/httpx/_client.py", line 1361, in request

    response = await self.send(

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/vcr/stubs/httpx_stubs.py", line 124, in _async_vcr_send

    vcr_request, response = _shared_vcr_send(cassette, real_send, *args, **kwargs)

  File "/home/oqt/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-UcUQmuRU-py3.8/lib/python3.8/site-packages/vcr/stubs/httpx_stubs.py", line 79, in _shared_vcr_send

    raise CannotOverwriteExistingCassetteException(cassette=cassette, failed_request=vcr_request)

vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/opt/oqt/tests/integrationtests/fixtures/vcr_cassettes/mock.json') in your current record mode ('none').

No match for the request (<Request (POST) https://api.ohsome.org/v1/elements/count>) was found.

No similar requests, that have not been played, found.

DEBUG    vcr.cassette:cassette.py:66 Exiting context for cassette at /opt/oqt/tests/integrationtests/fixtures/vcr_cassettes/mock.json.


---------- coverage: platform linux, python 3.8.11-final-0 -----------

Coverage XML written to file coverage.xml


=========================== short test summary info ============================

FAILED tests/integrationtests/test_oqt.py::TestOqt::test_create_all_indicators

=================== 1 failed, 118 passed in 64.61s (0:01:04) ===================

script returned exit code 1
```

**Review by joker234 — COMMENTED** [review]
> `VCR_RECORD_MODE=none pytest test_oqt.py`

Didn't work for me, as it should be when CI fails as well.

I made several commits. 
* rebased to the current main
* 874446cc70f9d52e66d6df51bd22818e0585447d increases the general timeouts. This was necessary as the test you wrote perform a lot of requests to the ohsome API and you don't always get an answer in time.
* 992fa309a7377a121c47d36c64c31cd368664a79 swaps the two decorators before the test as the VCR decorators have to come directly before the functions. Otherwise they can't find the correct filenames.
* db89e324b8ccdf56a8b63ba9ebed94fc5409c360 recreates the whole test_oqt cassettes. With the new test you're doing **a lot** of new requests (to create all indicators). I'm not sure if there isn't a better solution for that.

File: workers/ohsome_quality_analyst/oqt.py [hunk 1]
```diff
@@ -229,31 +230,37 @@ async def create_all_indicators(force: bool = False) -> None:
...
+                        fid,
+                        force,
+                        semaphore,
                     )
-                    continue
-                else:
-                    raise (error)
+                )
+            )
+    return await asyncio.gather(*tasks)
```
Comment (joker234) [inline workers/ohsome_quality_analyst/oqt.py:263]: Why the `return`?

Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/oqt.py]: There is no reason really.. I removed the return statement.

**Review by joker234 — APPROVED** [review]

**Pull request merged**